File: Create Zip/javascript/createzip.js
/*
 * IPWorks ZIP 2022 JavaScript Edition - Demo Application
 *
 * Copyright (c) 2023 /n software inc. - All rights reserved. - www.nsoftware.com
 *
 */
 
const readline = require("readline");
const ipworkszip = require("@nsoftware/ipworkszip");

if(!ipworkszip) {
  console.error("Cannot find ipworkszip.");
  process.exit(1);
}
let rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout
});

main();

async function main() {
  const zip = new ipworkszip.zip();

  prompt('archivefile','Please enter the name of the zip file to create', ':', 'test.zip');

  rl.on('line', async function(line)
  {
    switch(lastPrompt)
    {
      case 'archivefile':
      {
        if(line === '')
          zip.setArchiveFile(lastDefault);
        else
          zip.setArchiveFile(line);

        prompt('recurse','Recurse?',':','n');
        break;
      }
      case 'recurse':
      {
        if(line !== '')
          zip.setRecurseSubdirectories(true);

        prompt('folder','Please enter the path of the directory to compress',':','./*');
        break;
      }
      case 'folder':
      {
        if(line === '')
          zip.includeFiles(lastDefault);
        else
          zip.includeFiles(line);

        console.log('Compressing ...');
        try {
          await zip.compress();
          console.log('Compressed directory');
          process.exit();
        }
        catch (e) {
          console.log(e);
          process.exit();
        }
      }
    }
  });
}

function prompt(promptName, label, punctuation, defaultVal)
{
  lastPrompt = promptName;
  lastDefault = defaultVal;
  process.stdout.write(`${label} [${defaultVal}] ${punctuation} `);
}

File: Create Zip/cpp/createzip.cpp
/*
 * IPWorks ZIP 2022 C++ Edition - Sample Project
 *
 * This sample project demonstrates the usage of IPWorks ZIP in a 
 * simple, straightforward way. This is not intended to be a complete 
 * application. Error handling and other checks are simplified for clarity.
 *
 * Copyright (c) 2023 /n software inc. www.nsoftware.com
 */

#include <stdio.h>
#include <string.h>
#include <stdlib.h>
#include "../../include/ipworkszip.h"
#define LINE_LEN 100

int main()
{

	int			ret_code = 0;
	Zip zip;
	char		buffer[LINE_LEN+1];

	printf("**********************************************************\n");
	printf("* This is a demo of the Zip component                    *\n");
	printf("**********************************************************\n\n");

	printf("Please enter the name of the archive to create [test.zip]: " );
	fgets(buffer,LINE_LEN,stdin);
	buffer[strlen(buffer)-1] = '\0';
	if (strlen(buffer) == 0) strcat(buffer,"test.zip");
	zip.SetArchiveFile(buffer);

	printf("Recurse Sub-Directories? [N]: ");
	fgets(buffer,LINE_LEN,stdin);
	buffer[strlen(buffer)-1] = '\0';
	if (strlen(buffer) == 0) strcat(buffer,"N");

	if(strcmp("n",buffer) == 0 || strcmp("N",buffer) == 0)
		zip.SetRecurseSubdirectories(false);

	printf("Please enter the path and filemask of the directory to compress [*.cpp]: ");
	fgets(buffer,LINE_LEN,stdin);
	buffer[strlen(buffer)-1] = '\0';
	if (strlen(buffer) == 0) strcat(buffer,"*.cpp");
	zip.IncludeFiles(buffer);

	printf("\nCompressing ...\n\n");
	ret_code = zip.Compress();

	if(!ret_code)
		printf("Completed.\n\n");
	else
		printf("Error compressing: [%i] %s\n",ret_code,zip.GetLastError());


	fprintf(stderr, "\npress <return> to continue...\n");
	getchar();

	return ret_code;
}


